### PR TITLE
[kermi] Initial contribution

### PR DESCRIPTION
**Pull request by col-panic**
# Kermi Binding

- [kermi] Initial contribution - read-only values

# Description

This binding connects to [Kermi x-center controller](https://www.kermi.com/en/de/indoor-climate/products/heat-pumps-and-storage/x-center-controller/) for heat pumps.

### CODE REVIEW

**Comment by jlaur** [comment]
@col-panic - thanks for your contribution! Until this PR will picked up for review, you could already have a look at the checkstyle warnings. See target/code-analysis/report.html.

**Comment by col-panic** [comment]
I fixed some of those. Some of them don't make sense to me. Why is lang3.Stringutils provided, but should not be used?

**Comment by jlaur** [comment]
> Why is lang3.Stringutils provided, but should not be used?

See #7722.

File: bundles/org.openhab.binding.kermi/src/main/java/org/openhab/binding/kermi/internal/api/KermiHttpUtil.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                    if (e.getCause() instanceof InterruptedException) {
+                        throw (InterruptedException) e.getCause();
+                    }
+
+                    if (e.getCause() instanceof ExecutionException) {
+                        ExecutionException iex = (ExecutionException) e.getCause();
+                        if (iex != null && iex.getCause() instanceof HttpResponseException) {
+                            HttpResponseException hre = (HttpResponseException) iex.getCause();
+                            if (401 == hre.getResponse().getStatus()) {
+                                logger.debug("Perform login");
```
Comment (lsiepel) [inline bundles/org.openhab.binding.kermi/src/main/java/org/openhab/binding/kermi/internal/api/KermiHttpUtil.java:108]: better to move this debugging inside the `performlogin` method itself as that method is responsible for such a task.

File: bundles/org.openhab.binding.kermi/src/main/java/org/openhab/binding/kermi/internal/api/KermiHttpUtil.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                    }
+                    return result;
+                }
+
+                if (attemptCount >= 3) {
+                    logger.debug("Failed connecting to {} after {} attempts.", url, attemptCount, lastException);
+                }
+
+                logger.debug("HTTP error on attempt #{} {}", attemptCount, url);
+                Thread.sleep(500 * attemptCount);
```
Comment (lsiepel) [inline bundles/org.openhab.binding.kermi/src/main/java/org/openhab/binding/kermi/internal/api/KermiHttpUtil.java:129]: Are you sure you want to block the thread? why not use the scheduler? If this code is called on binding initilization, this is a no go. (i can check, but have to go now)

**Review by lsiepel — COMMENTED** [review]
Partly review, hope this is usefull to you. As I'm no maintaner consider this just as suggestions and to speed up the proces.

File: bundles/org.openhab.binding.kermi/pom.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.openhab.addons.bundles</groupId>
+    <artifactId>org.openhab.addons.reactor.bundles</artifactId>
+    <version>4.1.0-SNAPSHOT</version>
```
Comment (wborn) [inline bundles/org.openhab.binding.kermi/pom.xml:10]:
```suggestion
    <version>4.2.0-SNAPSHOT</version>
```

**Comment by lsiepel** [comment]
@col-panic are you able to proceed with the first comments and fix the build? If not, let me know and see what we can do to get this fixed.

**Comment by col-panic** [comment]
Hey @lsiepel feel free to perform changes. I've been using the code for a while for myself. At the moment I do not see a time slot to continue! It would be best, however, if someone could co-test the code if he has the same device.

**Comment by col-panic** [comment]
As of the current info, this PR will be closed in the future in favour of https://github.com/openhab/openhab-addons/pull/16329

**Pull request closed without merging**